Hard to reproduce failure in tests/devices_test/device_packages_test.py - DevicePackagesTestCase

With Python3 I sometimes get this error:
```
======================================================================
FAIL: testPackages (device_packages_test.DevicePackagesTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/var/www/html/blivet/tests/devices_test/device_packages_test.py", line 26, in testPackages
    self.assertListEqual(packages, list(set(packages)))
AssertionError: Lists differ: ['cryptsetup', 'mdadm'] != ['mdadm', 'cryptsetup']

First differing element 0:
cryptsetup
mdadm

- ['cryptsetup', 'mdadm']
+ ['mdadm', 'cryptsetup']
```
I've managed to isolate the issue down to the conversion from list to set and then back to list.

Initially packages is ['cryptsetup', 'mdadm'], then set(packages) becomes {'mdadm', 'cryptsetup'} which later becomes the list ['mdadm', 'cryptsetup'] and the assert fails. Converting to set under Python3 sometimes reverses the order of the elements but not always. Any ideas how to reproduce reliably or why that happens ?

--- a/tests/devices_test/device_packages_test.py
+++ b/tests/devices_test/device_packages_test.py
@@ -20,7 +20,7 @@ class DevicePackagesTestCase(unittest.TestCase):
         packages = luks.packages
 
         # no duplicates in list of packages
-        self.assertListEqual(packages, list(set(packages)))
+        self.assertEqual(len(packages), len(set(packages)))
 
         # several packages that ought to be included are
         for package in dev1.packages + dev2.packages + dev.packages:

